Sources/Jetworking/Client/Client.swift

```diff
@@ -206,14 +208,23 @@ public final class Client {
         guard checkForValidDownloadURL(url) else { return nil }
 
         let request: URLRequest = .init(url: url)
-        let task = downloadExecutor.download(request: request)
-        task.flatMap {
-            executingDownloads[$0.identifier] = DownloadHandler(
-                progressHandler: progressHandler,
-                completionHandler: completion
-            )
+
+        // Performs completion handler immediately with cached URL if available,
+        // otherwise executes the download request
+        if let url = sessionCache.query(URL.self, for: request) {
```

Comment:
I think we should provide a possibility to bypass the cached response e.g. if there is a need to refresh some data.

Reply:
Done